packages/@react-stately/color/src/useHexColorFieldState.ts

```diff
@@ -0,0 +1,121 @@
+import {useCallback, useMemo, useState} from 'react';
+import {useControlledState} from '@react-stately/utils';
+
+export interface HexColorFieldState extends NumberFieldState {
+  colorValue: Color,
+  setColorValue: (color: Color) => void,
+  setInputValue: (value: string) => void
+}
+
+/**
+ * Manages state for a color object, represented in hex, 
+ * and provides methods for manipulating the color value.
+ */
+export function useHexColorFieldState(
+  props: HexColorFieldProps
+): HexColorFieldState {
+  const {
+    minValue = '#000000',
+    maxValue = '#ffffff',
+    step = 1,
+    value,
+    defaultValue,
+    onChange,
+    validationState
+  } = props;
+
+  const [colorInputValue, setColorInputValue] = useControlledState<ColorInput>(value, defaultValue || minValue, onChange);
+
+  const minColor = Color.parse(minValue);
+  const maxColor = Color.parse(maxValue);
+  const minColorInt = minColor.toHexInt();
+  const maxColorInt = maxColor.toHexInt();
+
+  const colorValue = useMemo(() => {
+    let validColor;
+    try {
+      const color = Color.parse(colorInputValue);
+      const colorInt = color.toHexInt();
+      if (colorInt <= maxColorInt && colorInt >= minColorInt) {
+        validColor = color;
+      }
+    } catch (err) {
+      // ignore
+    }
+    return validColor;
+  }, [colorInputValue, minColorInt, maxColorInt]);
+
+  const [inputValue, setInputValue] = useState(colorValue ? colorValue.toString('hex') : '');
+
+  const increment = () => {
+    setColorInputValue((previousValue) => {
+      const colorInt = Color.parse(previousValue).toHexInt();
+      const newValue = `#${Math.min(colorInt + step, maxColorInt).toString(16).padStart(6, '0').toUpperCase()}`;
+      const newColor = new Color(newValue);
+      setInputValue(newValue);
+      return newColor;
+    });
+  };
+
+  const incrementToMax = useCallback(() => {
+    if (maxValue != null) {
+      setColorInputValue(maxColor);
+      setInputValue(maxColor.toString('hex'));
+    }
+  }, [maxValue, maxColor, setColorInputValue, setInputValue]);
+
+  const decrement = () => {
+    setColorInputValue((previousValue) => {
+      const colorInt = Color.parse(previousValue).toHexInt();
+      const newValue = `#${Math.max(colorInt - step, minColorInt).toString(16).padStart(6, '0').toUpperCase()}`;
+      const newColor = new Color(newValue);
+      setInputValue(newValue);
+      return newColor;
+    });
+  };
+
+  const decrementToMin = useCallback(() => {
+    if (minValue != null) {
+      setColorInputValue(minColor);
+      setInputValue(minColor.toString('hex'));
+    }
+  }, [minValue, minColor, setColorInputValue, setInputValue]);
+
+  const setColorValue = (color: Color) => {
+    setColorInputValue(color);
+  };
+
+  const commitInputValue = () => {
+    setInputValue(colorValue.toString('hex'));
+  };
+
+  return {
+    colorValue,
+    setColorValue,
+    inputValue,
+    setInputValue,
+    commitInputValue,
```

Comment:
I feel like `inputValue` + its other methods should be in aria instead of stately. Stately should concern itself only on the underlying data structure (`Color`) and how it's represented as text in the UI/input field is up to the aria hook. 

Is this is why we call it `useHexColorFieldState` instead of `useHexColorState`? Coz it also covers states more specific to the input field like `inputValue`?

Reply:
@snowystinger 
Revisit this after making changes in setColorValue